systemd/init

```diff
@@ -0,0 +1,52 @@
+#!/bin/busybox sh
+
+# Init for Wolfi OS
+# - Environment variables:
+#     VERBOSE=1 for debugging
+#     CMDLINE=/path/to/file to override /proc/cmdline, useful for testing
+# - Mount /proc, /sys, /dev if not already mounted
+# - Allow for override of path to /init with init=/path/to/init on kernel /proc/cmdline
+# - Allow for uuid or device path for root filesystem with root= on kernel /proc/cmdline
+# - Mount root, if specified and found, switch_root to it, and run systemd as init
+# - Otherwise, try to run systemd as init in the initrd
+
+# Verbose debugging
+info() { [ "$VERBOSE" = "1" ] && printf "INFO: $@\n" 1>&2; }
+
+# Override /proc/cmdline path as environment variable, useful for testing
+[ -z "$CMDLINE" ] && CMDLINE="/proc/cmdline"
+info "cmdline = [$(cat $CMDLINE)]"
+
+# Setup our system mounts if needed
+mkdir -p /proc /sys /dev
+findmnt /proc >/dev/null 2>&1 || mount -t proc none /proc
+findmnt /sys >/dev/null  2>&1 || mount -t sysfs none /sys
+findmnt /dev >/dev/null  2>&1 || mount -t devtmpfs none /dev
```

Comment:
why would /proc or /sys or /dev ever be mounted?

Reply:
During a real, actual, fresh boot, they should not.  But I was also doing some testing in a container, just trying to get the shell logic right.  And in the container, they are already mounted.